code/game/machinery/slotmachine.dm

```diff
@@ -118,4 +126,70 @@
 	. = TRUE
 	if(!I.tool_use_check(user, 0))
 		return
+	// This prevents the emagged machine to be moved while someone is playing on it.
+	if(emagged_game_in_progress)
+		return
 	default_unfasten_wrench(user, I)
+
+// Emag behaviour below. When the machine is emagged, it'll stun and anchor its user, spin them, then throw them away.
+// With a chance of "emagged_win_chance", the machine resets its emagged state and throws money at the user.
+/obj/machinery/economy/slot_machine/emag_act(user)
+	if(emagged)
+		to_chat(user, "<span class='notice'>[src] is unresponsive. It is probably already modified.</span>")
+		return
+	playsound(loc, 'sound/effects/sparks4.ogg', 75, 1)
+	emagged = TRUE
+	to_chat(user, "<span class='notice'>You engage the reverse-gripping mechanism on the machine's handle.</span>")
+	log_game("[key_name(user)] emagged [src]")
+
+// The spinning and throwing away is handled here, with a possible call to winning
+/obj/machinery/economy/slot_machine/proc/emagged_spinning(mob/living/user)
+	to_chat(user, "<span class='danger'>As you grip the handle of the machine, it grips back at you, and starts to wildly spin you around!</span>")
+	user.SpinAnimation(speed = 2, loops = 6)
+	emagged_game_in_progress = TRUE
+
+	// Make sure the machine is anchored to avoid people cheesing it
+	if(!anchored)
+		atom_say("Deploying safety bolts.")
+		anchored = TRUE
+
+	// Stun them, there is no escape from this unless you get teleported
+	user.anchored = TRUE
+	user.SetStunned(1.2 SECONDS, TRUE)
+
+	// No cheesing with buckling ourselves, this spinning is too fast for seatbelts
+	if(user.buckled)
+		user.buckled.unbuckle_mob(user, force = TRUE)
+
+	// Check if the machine and the user are still next to each other
+	if(!do_after(user, delay = 1.2 SECONDS, target = src, use_default_checks = FALSE))
+		user.anchored = FALSE
```

Comment:
the unanchor being gated behind a check scares me, how does the user get unanchored if this check fails?

Reply:
this ☝️

Reply:
They get unanchored regardless, this is a `!do_after`, meaning that this unanchoring happens _when_ the check fails. (And also when it succeeds, a few lines below).

You can admin-teleport them, throw them away with build mode, hell, delete the slot machine while this do_after is in progress - it'll fail and the user will get unanchored.